### ISSUE TITLE
New Google Chrome typeof "bigint" marked as invalid

### ISSUE THREAD

**Issue by jcubic** [eslint/eslint#10574]
**Tell us about your environment**

* **ESLint Version:** v4.19.1
* **Node Version:** v10.4.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

"parserOptions": {
      "ecmaVersion": 6,
      "sourceType": "script",
      "ecmaFeatures": {}
    },

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  "eslintConfig": {
    "parserOptions": {
      "ecmaVersion": 6,
      "sourceType": "script",
      "ecmaFeatures": {}
    },
    "env": {
      "browser": true,
      "jest": true,
      "node": true
    },
    "globals": {
      "Promise": true
    },
    "rules": {
      "eqeqeq": "error",
      "curly": "error",
      "no-unreachable": "error",
      "valid-typeof": "error",
      "no-unexpected-multiline": "error",
      "no-regex-spaces": "error",
      "no-irregular-whitespace": "error",
      "no-invalid-regexp": "error",
      "no-inner-declarations": "error",
      "no-func-assign": "error",
      "no-extra-semi": "error",
      "no-extra-boolean-cast": "error",
      "no-debugger": "error",
      "no-dupe-args": "error",
      "no-dupe-keys": "error",
      "no-duplicate-case": "error",
      "no-empty-character-class": "error",
      "no-ex-assign": "error",
      "array-callback-return": "error",
      "no-case-declarations": "error",
      "guard-for-in": "error",
      "no-caller": "error",
      "no-eval": "error",
      "no-extend-native": "error",
      "no-extra-bind": "error",
      "no-fallthrough": "error",
      "no-global-assign": "error",
      "no-implicit-globals": "error",
      "no-implied-eval": "error",
      "no-multi-spaces": "error",
      "no-new-wrappers": "error",
      "no-redeclare": "error",
      "no-self-assign": "error",
      "no-return-assign": "error",
      "no-self-compare": "error",
      "no-throw-literal": "error",
      "no-unused-labels": "error",
      "no-useless-call": "error",
      "no-useless-escape": "error",
      "no-void": "error",
      "no-with": "error",
      "radix": "error",
      "wrap-iife": [
        "error",
        "inside"
      ],
      "yoda": [
        "error",
        "never"
      ],
      "no-catch-shadow": "error",
      "no-delete-var": "error",
      "no-label-var": "error",
      "no-undef-init": "error",
      "no-unused-vars": "error",
      "no-undef": "error",
      "comma-style": [
        "error",
        "last"
      ],
      "comma-dangle": [
        "error",
        "never"
      ],
      "comma-spacing": [
        "error",
        {
          "before": false,
          "after": true
        }
      ],
      "computed-property-spacing": [
        "error",
        "never"
      ],
      "eol-last": [
        "error",
        "always"
      ],
      "func-call-spacing": [
        "error",
        "never"
      ],
      "key-spacing": [
        "error",
        {
          "beforeColon": false,
          "afterColon": true,
          "mode": "strict"
        }
      ],
      "max-len": [
        "error",
        85
      ],
      "max-statements-per-line": "error",
      "new-parens": "error",
      "no-array-constructor": "error",
      "no-lonely-if": "error",
      "no-mixed-spaces-and-tabs": "error",
      "no-multiple-empty-lines": "error",
      "no-new-object": "error",
      "no-tabs": "error",
      "no-trailing-spaces": "error",
      "no-whitespace-before-property": "error",
      "object-curly-spacing": [
        "error",
        "never"
      ],
      "space-before-blocks": "error",
      "keyword-spacing": [
        "error",
        {
          "before": true,
          "after": true
        }
      ],
      "space-in-parens": [
        "error",
        "never"
      ],
      "space-infix-ops": "error",
      "space-before-function-paren": [
        "error",
        "never"
      ],
      "complexity": [
        "error",
        {
          "max": 26
        }
      ],
      "indent": [
        "error",
        4,
        {
          "SwitchCase": 1
        }
      ],
      "linebreak-style": [
        "error",
        "unix"
      ],
      "semi": [
        "error",
        "always"
      ]
    }
  },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
    function LNumber(n) {
        if (n instanceof LNumber) {
            return n;
        }
        // prevent infite loop https://github.com/indutny/bn.js/issues/186
        if (n === null) {
            n = 0;
        }
        if (typeof this !== 'undefined' && this.constructor !== LNumber ||
            typeof this === 'undefined') {
            return new LNumber(n);
        }
        if (typeof BigInt !== 'undefined' && typeof n !== 'bigint') {
            this.value = BigInt(n);
        } else if (typeof BN !== 'undefined' && !(n instanceof BN)) {
            this.value = new BN(n);
        } else {
            this.value = n;
        }
    }
```

```bash
./node_modules/.bin/eslint src/lips.js spec/lips.spec.js
```

**What did you expect to happen?** it should not give error

**What actually happened? Please include the actual, raw output from ESLint.** it give error:

542:59  error  Invalid typeof comparison value  valid-typeof

when parsing typeof "bigint" added by Google Chrome

**Comment by platinumazure** [comment]
Hi @jcubic, thanks for the issue!

Bigint is currently stage 3. So our default parser and rules currently do not support it. (See https://github.com/eslint/eslint/blob/master/README.md#what-about-experimental-features).

Your best chance at this point is to use babel-eslint (to parse literals), as well as to see if there is a rule in eslint-plugin-babel which augments valid-typeof to support big integers.

**Comment by jcubic** [comment]
Thanks @platinumazure will use the plugin then, right now I've disabled this rule.
